src/Symfony/Component/Yaml/Inline.php

```diff
@@ -466,14 +468,16 @@ private static function evaluateScalar($scalar, $references = array())
                         return '0' == $scalar[1] ? octdec($scalar) : (((string) $raw === (string) $cast) ? $cast : $raw);
                     case is_numeric($scalar):
                     case preg_match(self::getHexRegex(), $scalar):
+                        $scalar = str_replace('_', '', $scalar);
+
                         return '0x' === $scalar[0].$scalar[1] ? hexdec($scalar) : (float) $scalar;
                     case '.inf' === $scalarLower:
                     case '.nan' === $scalarLower:
                         return -log(0);
                     case '-.inf' === $scalarLower:
                         return log(0);
-                    case preg_match('/^(-|\+)?[0-9,]+(\.[0-9]+)?$/', $scalar):
```

Comment:
I don't think I saw anywhere that the `,` was authorized though ? I kept it, but nothing in the specs says it should be used as a separator (And as @nicolas-grekas noticed, `_123_456_` should not be a valid number, so shouldn't `,123,456,` be also considered invalid or a string ?